.github/workflows/upload_binary.yml

```diff
@@ -0,0 +1,51 @@
+name: Upload self-contained binaries
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.7]
+        os: [ubuntu-16.04, windows-2019]
+        include:
+          - os: windows-2019
+            pathsep: ";"
+            executable_suffix: ".exe"
+            executable_mime: "application/vnd.microsoft.portable-executable"
+          - os: ubuntu-16.04
+            pathsep: ":"
+            executable_suffix: ".elf"
+            executable_mime: "application/x-executable"
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip wheel setuptools
+          python -m pip install .
+          python -m pip install pyinstaller
+
+      - name: Build binary
+        run: |
+          python -m PyInstaller -F --name black${{ matrix.executable_suffix }} --add-data 'src/blib2to3${{ matrix.pathsep }}blib2to3' src/black/__main__.py
```

Comment:
Maybe I'm misunderstanding  how this works, but why is `blib2to3` in the path twice? (`src/blib2to3${{ matrix.pathsep }}blib2to3`)

Reply:
`src/blib2to3` is the source of the data to include and `blib2to3` is the destination path. PyInstaller creates an one-folder bundle that contains an executable so Pyinstaller needs to know where to put the extra data in the bundle directory. `dist` is `blib2to3` and not `src/blib2to3` because PyInstaller hijacks calls to `pkgutil.get_data`. The one file executable PyInstaller can create is really just a bootloader plus an embedded archive that is decompressed and extracted on execution. 

A better explanation is available here: https://pyinstaller.readthedocs.io/en/stable/spec-files.html#adding-data-files